regenerate.sh

```diff
@@ -34,43 +34,56 @@ rm -f $(find . -name '*.pb.go' | grep -v 'grpc_testing_not_regenerate')
 echo "go install google.golang.org/protobuf/cmd/protoc-gen-go"
 (cd test/tools && go install google.golang.org/protobuf/cmd/protoc-gen-go)
 
-echo "go install cmd/protoc-gen-go-grpc"
-(cd cmd/protoc-gen-go-grpc && go install .)
+ echo "go install cmd/protoc-gen-go-grpc"
+ (cd cmd/protoc-gen-go-grpc && go install .)
 
-echo "git clone https://github.com/grpc/grpc-proto"
-git clone --quiet https://github.com/grpc/grpc-proto ${WORKDIR}/grpc-proto
+ echo "git clone https://github.com/grpc/grpc-proto"
+ git clone --quiet https://github.com/grpc/grpc-proto "${WORKDIR}"/grpc-proto
 
-echo "git clone https://github.com/protocolbuffers/protobuf"
-git clone --quiet https://github.com/protocolbuffers/protobuf ${WORKDIR}/protobuf
+ echo "git clone https://github.com/protocolbuffers/protobuf"
+ git clone --quiet https://github.com/protocolbuffers/protobuf "${WORKDIR}"/protobuf
 
-# Pull in code.proto as a proto dependency
-mkdir -p ${WORKDIR}/googleapis/google/rpc
-echo "curl https://raw.githubusercontent.com/googleapis/googleapis/master/google/rpc/code.proto"
-curl --silent https://raw.githubusercontent.com/googleapis/googleapis/master/google/rpc/code.proto > ${WORKDIR}/googleapis/google/rpc/code.proto
+ # Pull in code.proto as a proto dependency
+ mkdir -p "${WORKDIR}"/googleapis/google/rpc
+ echo "curl https://raw.githubusercontent.com/googleapis/googleapis/master/google/rpc/code.proto"
+ curl --silent https://raw.githubusercontent.com/googleapis/googleapis/master/google/rpc/code.proto > "${WORKDIR}"/googleapis/google/rpc/code.proto
 
-mkdir -p ${WORKDIR}/out
+source ./protoc_installer.sh
+download_protoc
+
+export PATH="$PATH:$GOBIN"
```

Comment:
why is adding the GOBIN necessary here?

Reply:
Yeah as we downloading the zip and the installing it under directory WORKDIR, we would need to add the binary path to GOPATH so when we run protoc it points to the one we installed.